File: Php/core/Validation/SimpleValidator.php
<?php
namespace core\Validation{

	abstract class SimpleValidator extends SingleValidator {

		protected $subject;
		protected $message;

		public function __construct($subject, $message){
			$this->subject = $subject;
			$this->message = $message;
		}

		public function validate(){
			$this->validateSubject();
		}

		abstract function validateSubject();
		
	}

}

File: Web/Validators/SimpleValidator.cs
namespace Web.Validators
{
    public abstract class SimpleValidator : SingleValidator
    {
	    protected readonly string Subject;
	    protected readonly string Message;

	    public SimpleValidator(string subject, string message)
	    {
	        Subject = subject;
	        Message = message;
	    }

	    protected override void Validate(){
			ValidateSubject();
		}

	    protected abstract bool ValidateSubject();
		
	}

}